c-log/clog.go

```diff
@@ -0,0 +1,218 @@
+func (l *Logger) Error(title string) {
+	l.ErrorD(title, m{})
+}
+
+// Critical takes a string and logs with LogLevel = Critical
+func (l *Logger) Critical(title string) {
+	l.CriticalD(title, m{})
+}
+
+// Counter takes a string and logs with LogLevel = Info, type = counter, and value = 1
+func (l *Logger) Counter(title string) {
+	l.CounterD(title, 1, m{}) // Defaults to a value of 1
+}
+
+// Gauge takes a string and integer value. It logs with LogLevel = Info, type = gauge, and value = value
+func (l *Logger) Gauge(title string, value int) {
+	l.GaugeD(title, value, m{})
+}
+
+// DebugD takes a string and data map. It logs with LogLevel = Debug
+func (l *Logger) DebugD(title string, data map[string]interface{}) {
+	data["title"] = title
+	l.logWithLevel(Debug, data)
+}
+
+// InfoD takes a string and data map. It logs with LogLevel = Info
+func (l *Logger) InfoD(title string, data map[string]interface{}) {
+	data["title"] = title
+	l.logWithLevel(Info, data)
+}
+
+// WarnD takes a string and data map. It logs with LogLevel = Warning
+func (l *Logger) WarnD(title string, data map[string]interface{}) {
+	data["title"] = title
+	l.logWithLevel(Warning, data)
+}
+
+// ErrorD takes a string and data map. It logs with LogLevel = Error
+func (l *Logger) ErrorD(title string, data map[string]interface{}) {
+	data["title"] = title
+	l.logWithLevel(Error, data)
+}
+
+// CriticalD takes a string and data map. It logs with LogLevel = Critical
+func (l *Logger) CriticalD(title string, data map[string]interface{}) {
+	data["title"] = title
+	l.logWithLevel(Critical, data)
+}
+
+// CounterD takes a string, value, and data map. It logs with LogLevel = Info, type = counter, and value = value
+func (l *Logger) CounterD(title string, value int, data map[string]interface{}) {
+	data["title"] = title
+	data["value"] = value
+	data["type"] = "counter"
+	l.logWithLevel(Info, data)
+}
+
+// GaugeD takes a string, value, and data map. It logs with LogLevel = Info, type = gauge, and value = value
+func (l *Logger) GaugeD(title string, value int, data map[string]interface{}) {
+	data["title"] = title
+	data["value"] = value
+	data["type"] = "gauge"
+	l.logWithLevel(Info, data)
+}
+
+// Actual logging. Handles whether to output based on log level and
+// unifies the passed in data with the stored globals
+func (l *Logger) logWithLevel(logLvl LogLevel, data map[string]interface{}) {
+	if logLvl < l.logLvl {
+		// No log output
+		return
+	}
+	data["level"] = logLvl.String()
+	for key, value := range l.globals {
+		if _, ok := data[key]; ok {
+			// Values in the data map override the globals
+			continue
+		}
+		data[key] = value
+	}
+
+	logString := l.formatter(data)
+	l.logWriter.Println(logString)
+}
+
+// New creates a *clog.Logger. Default values are Debug LogLevel, kayvee Formatter, and std.err output.
+func New(source string) *Logger {
+	logObj := Logger{}
+	var logLvl LogLevel
+	strLogLvl := os.Getenv("LOG_LEVEL_CONFIG")
```

Comment:
nit: maybe `KAYVEE_LOG_LEVEL`? seems unusual to name a config var with `CONFIG` in the name